sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +41,45 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// Registers an <see cref="IPersistableModel{T}"/> proxy to be used when reading or writing a model.
+    /// </summary>
+    /// <param name="type"> The <see cref="Type"/> that will be proxied when asked to be read or written. </param>
+    /// <param name="proxy"> The <see cref="IPersistableModel{T}"/> proxy that will be used to read or write the model. </param>
+    public void AddProxy<T>(Type type, IPersistableModel<T> proxy)
+    {
+        //multiple of same type?
```

Comment:
Did you decide on a story for this?  Should it throw an exception for now if a proxy already exists for a given type?